palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/request/OriginalRequest.java

```diff
@@ -0,0 +1,144 @@
+ * Note there are two classes that effectively represent the same data but represent a different stage of the process.
+ * uk.gov.gchq.palisade.service.palisade.request.OriginalRequest is the client request that has come into the Palisade Service.
+ * uk.gov.gchq.palisade.service.user.request.UserRequest is the input for the User Service.
+ */
+
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public final class OriginalRequest {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private final String userId;  //Unique identifier for the user
+    private final String resourceId;  //Resource that that is being asked to access
+    private final JsonNode context; //Relevant context information about the request.
+
+
+    @JsonCreator
+    private OriginalRequest(
+            final @JsonProperty("userId") String userId,
+            final @JsonProperty("resourceId") String resourceId,
+            final @JsonProperty("context") JsonNode context) {
+
+        Assert.notNull(userId, "User cannot be null");
+        Assert.notNull(resourceId, "Resource cannot be null");
+        Assert.notNull(context, "Context cannot be null");
+
+        this.userId = userId;
+        this.resourceId = resourceId;
+        this.context = context;
+    }
+
+    @Generated
+    public String getUserId() {
+        return userId;
+    }
+
+    @Generated
+    public String getResourceId() {
+        return resourceId;
+    }
+
+    @Generated
+    public Context getContext() throws JsonProcessingException {
+        return MAPPER.treeToValue(context, Context.class);
+    }
+
+    @Override
+    @Generated
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof OriginalRequest)) {
+            return false;
+        }
+        OriginalRequest that = (OriginalRequest) o;
+        return userId.equals(that.userId) &&
+                resourceId.equals(that.resourceId) &&
+                context.equals(that.context);
+    }
+
+    @Override
+    @Generated
+    public int hashCode() {
+        return Objects.hash(userId, resourceId, context);
+    }
+
+    @Override
+    @Generated
+    public String toString() {
+        return new StringJoiner(", ", OriginalRequest.class.getSimpleName() + "[", "]")
+                .add("userId='" + userId + "'")
+                .add("resourceId='" + resourceId + "'")
+                .add("context=" + context)
+                .add(super.toString())
+                .toString();
+    }
+
+    /**
+     * Builder class for the creation of instances of the OriginalRequest.  This is a variant of the Fluent Builder
+     * which will use the String or optionally JsonNodes for the components in the build.
+     */
+    public static class Builder {
+
+        public static IUser create() {
+            return user -> resource -> context ->
```

Comment:
SonarQube will complain about these missing types maybe?
(String user) -> (String resource) -> ....

Reply:
I think it is clever with this.   It picks up from the interfaces what the types have to be.